src/efi.rs

```diff
@@ -451,18 +452,45 @@ impl Component for Efi {
     }
 
     fn generate_update_metadata(&self, sysroot: &str) -> Result<ContentMetadata> {
-        let sysroot_path = Utf8Path::new(sysroot);
+        let sysroot_path = Path::new(sysroot);
+        let sysroot_dir = Dir::open_ambient_dir(sysroot_path, cap_std::ambient_authority())?;
+
+        if let Some(ostreeboot) = sysroot_dir
+            .open_dir_optional(ostreeutil::BOOT_PREFIX)
+            .context("Opening usr/lib/ostree-boot")?
+        {
+            let cruft = ["loader", "grub2"];
+            for p in cruft.iter() {
+                ostreeboot.remove_all_optional(p)?;
+            }
+
+            let efisrc = sysroot_path.join(ostreeutil::BOOT_PREFIX).join("efi/EFI");
+            if efisrc.exists() {
+                let files = WalkDir::new(&efisrc)
+                    .into_iter()
+                    .filter_map(Result::ok)
+                    .filter(|e| e.file_type().is_file())
+                    .map(|e| e.path().to_path_buf())
+                    .collect::<Vec<_>>();
+
+                if !files.is_empty() {
```

Comment:
On rawhide with latest grub2 & shim, there are existing empty directories `/usr/lib/ostree-boot/efi/EFI/{BOOT,fedora}`, this is because when installing `efi-filesystem`, `boot/` is transferred to `usr/lib/ostree-boot`, need to remove the directory `/usr/lib/ostree-boot/efi/EFI`. Let me know if I misunderstood this.
```
bash-5.3# rpm -ql efi-filesystem
/boot/efi
/boot/efi/EFI
/boot/efi/EFI/BOOT
/boot/efi/EFI/fedora
```